release/v1alpha1/types.go

```diff
@@ -0,0 +1,185 @@
+//          the release will be marked `Rejected`
+//   7) Publishes all results to the respective webpage
+//
+// Example:
+// ART:
+//   1) Publishes an update to the `ocp/4.9-art-latest` imagestream
+//
+// Release-controller:
+//   1) Creates a mirror named: `ocp/4.9-art-latest-2021-09-27-105859`
+//   2) Creates an OpenShift Release: `ocp/release:4.9.0-0.nightly-2021-09-27-105859`
+//   3) Launches: 4.9.0-0.nightly-2021-09-27-105859-aggregated-<name>-analysis-<count>
+//   4) Launches: 4.9.0-0.nightly-2021-09-27-105859-aggregated-<name>-aggregator
+//   5) Launches: 4.9.0-0.nightly-2021-09-27-105859-<name>
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type ReleasePayload struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec the inputs used to create the ReleasePayload
+	Spec ReleasePayloadSpec `json:"spec,omitempty"`
+
+	// Status is the current status of the ReleasePayload
+	Status ReleasePayloadStatus `json:"status,omitempty"`
+}
+
+// ReleasePayloadSpec has the information to represent a PromotionTest
+type ReleasePayloadSpec struct {
+	// ReleaseTag a pointer to the imagestreamtag in the TargetImageStream
+	ReleaseTag v1.ImageStreamTag `json:"releaseTag,omitempty"`
+
+	// SourceImageStream a pointer to the imagestream where this ReleasePayload was created from
+	SourceImageStream *v1.ImageStream `json:"source,omitempty"`
+
+	// TargetImageStream a pointer to the imagestream where this ReleasePayload will be located
+	TargetImageStream *v1.ImageStream `json:"target,omitempty"`
+}
+
+// ReleasePayloadStatus the status of all the promotion test jobs
+type ReleasePayloadStatus struct {
+	// Conditions communicates the state of the ReleasePayload.
+	//
+	// Supported conditions include PayloadCreated, Failed, Accepted, and Rejected.
+	//
+	// If PayloadCreated is false the ReleasePayload is waiting for a release image to be created and pushed to the
+	// TargetImageStream.  If PayloadCreated is true a release image has been created and pushed to the TargetImageStream.
+	// Verification jobs should begin and will update the status as they complete.
+	//
+	// If Failed is true a ReleasePayload image cannot be created for the given set of image mirrors
+	// This condition is terminal
+	//
+	// If Accepted is true the ReleasePayload has passed its verification criteria and can safely
+	// be promoted to an external location
+	// This condition is terminal
+	//
+	// if Rejected is true the ReleasePayload has failed one or more of its verification criteria
+	// The release-controller will take no more action in this phase, but a human may set the
+	// phase back to Ready to retry and the controller will attempt verification again.
```

Comment:
no they can't.  Humans don't set status.

Is this retry feature required?  If so, does it require a separate releasepayload instance?

Reply:
The current implementation does allow for manual, human, intervention, although this particular path is rarely used.  Manually, Accepting a release however is something that's done quite often.